core/chains/evm/txm/stuck_tx_detector.go

```diff
@@ -0,0 +1,115 @@
+package txm
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"time"
+
+	"github.com/ethereum/go-ethereum/common"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/logger"
+
+	"github.com/smartcontractkit/chainlink/v2/core/chains/evm/config/chaintype"
+	"github.com/smartcontractkit/chainlink/v2/core/chains/evm/txm/types"
+)
+
+type StuckTxDetectorConfig struct {
+	BlockTime             time.Duration
+	StuckTxBlockThreshold uint32
+	DetectionURL          string
+}
+
+type stuckTxDetector struct {
+	lggr      logger.Logger
+	chainType chaintype.ChainType
+	config    StuckTxDetectorConfig
+}
+
+func NewStuckTxDetector(lggr logger.Logger, chaintype chaintype.ChainType, config StuckTxDetectorConfig) *stuckTxDetector {
+	return &stuckTxDetector{
+		lggr:      lggr,
+		chainType: chaintype,
+		config:    config,
+	}
+}
+
+func (s *stuckTxDetector) DetectStuckTransaction(ctx context.Context, tx *types.Transaction) (bool, error) {
+	switch s.chainType {
+	// TODO: rename
+	case chaintype.ChainDualBroadcast:
+		result, err := s.dualBroadcastDetection(ctx, tx)
+		if result || err != nil {
+			return result, err
+		}
+		return s.timeBasedDetection(tx), nil
+	default:
+		return s.timeBasedDetection(tx), nil
+	}
+}
+
+func (s *stuckTxDetector) timeBasedDetection(tx *types.Transaction) bool {
```

Comment:
Would we be dropping the heuristic and custom detection logic we have for ZK chains or this just a temporary measure? If we are moving forward with this, I think we need to be careful with marking all inflight transactions as stuck here. If we broadcast multiple in quick succession, they can all have similar broadcast timestamps but just the lowest nonce one could be stuck holding up the rest of them.

Reply:
I was hoping we could merge the logics because currently the Stuck Tx Detector handles multiple txs at the same time and ideally I want us to process one transaction at a time.
As for this logic, you bring up a good point. However, the broadcasted txs with a higher nonce are already in the mempool so if the current nonce gets filled, they should be picked right away. I do agree though that we can use some of the more complex mechanisms of the Heuristic method to improve this, or somehow combine them.